torch/csrc/jit/passes/onnx/fold_if_node.cpp

```diff
@@ -0,0 +1,249 @@
+#include <torch/csrc/jit/passes/onnx/fold_if_node.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <torch/csrc/jit/passes/onnx/scalar_type_analysis.h>
+#include <torch/torch.h>
+
+#include <c10/util/Optional.h>
+#include <algorithm>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+// This function determines wheather If Node can be folded.
+static bool isStaticCondition(Node* node) {
+  TORCH_INTERNAL_ASSERT(
+      node->kind() == onnx::If || node->kind() == onnx::Not ||
```

Comment:
Put it in a function since it is called twice: 
node->kind() == onnx::If || node->kind() == onnx::Not ||
      node->kind() == onnx::Identity

Reply:
I think function would be an overhead